### PR TITLE
save query_results only when not nil

### LINKED ISSUE #949: Try exercises break on loading if last submission failed with no query_result
This can happen for example if connection to the runner server failed.

### PR DESCRIPTION
**Pull request by julian-berbel**
Fixes #949.

### CODE REVIEW

**Comment by flbulgarelli** [comment]
Could we add a test?

**Comment by julian-berbel** [comment]
yessir!

File: spec/models/interactive_spec.rb [hunk 1]
```diff
@@ -125,4 +125,19 @@
...
+    before do
+      expect(bridge).to receive(:run_try!).and_return({status: :errored, result: ''})
+    end
+
+    before do
+      interactive.submit_try!(user, query: 'foo')
+    end
+
+    it { expect(assignment.queries).to eq ['foo'] }
+    it { expect(assignment.query_results).to eq [] }
```
Comment (flbulgarelli) [inline spec/models/interactive_spec.rb]: Can adding a query without a query result led to inconsistencies?

Comment (julian-berbel) [reply spec/models/interactive_spec.rb]: https://github.com/mumuki/mumuki-laboratory/blob/06e42bb96b5d9ae429a872d3b0708b5c374590fc/app/models/assignment.rb#L30-L31

... yep.

File: app/models/submission/console_submission.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class ConsoleSubmission < Submission
+  required :try_evaluate_query!
+
+  def try_evaluate!(assignment)
+    format_query_result! try_evaluate_query!(assignment)
+  end
+
+  def format_query_result!(results)
+    results[:result] = I18n.t(:try_again) if results[:status] == :errored
```
Comment (flbulgarelli) [inline app/models/submission/console_submission.rb]: We should only show this message in aborted scenarios

**Pull request merged**